torch/lib/TH/generic/THTensorMath.c

```diff
@@ -424,11 +424,12 @@ void THTensor_(gather)(THTensor *tensor, THTensor *src, int dim, THLongTensor *i
 {
   int64_t elems_per_row, i, idx;
 
-  THArgCheck(THTensor_(nDimension)(src) == THTensor_(nDimension)(tensor), 2,
-             "Input tensor must have same dimensions as output tensor");
-  THArgCheck(dim < THTensor_(nDimension)(tensor), 3, "Index dimension is out of bounds");
+  // Reverse the order of checks to provide more helpful error messages
```

Comment:
Remove this comment. It’s not like there’s a canonical ordering on these checks so it’s only going to be confusing in the future.